app/services/fastly_vcl/whitelisted_params.rb

```diff
@@ -0,0 +1,67 @@
+module FastlyVCL
+  # Handles updates to our VCL snippet on Fastly that whitelists params
+  class WhitelistedParams
+    VCL_DELIMITER_START = "^(".freeze
+    VCL_DELIMITER_END = ")$".freeze
+    SNIPPET_NAME = "Whitelist certain querystring parameters".freeze
+    FILE_PARAMS = YAML.load_file("config/fastly/whitelisted_params.yml").freeze
+
+    class << self
+      def update
+        fastly = Fastly.new(api_key: ApplicationConfig["FASTLY_API_KEY"])
+        service = fastly.get_service(ApplicationConfig["FASTLY_SERVICE_ID"])
+        latest_version = service.version
+
+        snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                     latest_version.number,
+                                     SNIPPET_NAME)
+
+        current_params = params_to_array(snippet.content)
+
+        return unless params_updated?(current_params)
+
+        new_version = latest_version.clone
+        new_snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                         new_version.number,
+                                         SNIPPET_NAME)
+        new_snippet.content = build_content(FILE_PARAMS, new_snippet.content)
+        new_snippet.save!
+
+        new_version.activate!
+        log_params_diff_to_datadaog(current_params, new_version)
+        Rails.logger.info("Fastly updated to version #{new_version.number}.")
+      end
+
+      private
+
+      def params_updated?(current_params)
+        (current_params - FILE_PARAMS).any? || (FILE_PARAMS - current_params).any?
```

Comment:
If the current params in Fastly have things we don't have in our codebase I think we need to raise an error bc that would need to be addressed manually by an admin in order to get them back in sync.

Reply:
Ah, gotcha. I was sticking to the idea that the code is now the source of truth. So regardless if someone made an update in Fastly, it'd be overwritten to prevent needing admin intervention. If we raise an error, deployments wouldn't work until a Fastly admin fixed the issue. My original thinking is the addition of a whitelisted param in Fastly (most common update) wasn't worth blocking deployments pending Fastly admin updates.

I can raise an error though, I'll update this.